### PR TITLE
Networking V2: Update/Delete/ScheduleDHCP/RemoveDHCP on network agents

### LINKED ISSUE #1389: Networking V2: Agents API
Gophercloud should support Networking V2 service agents API.

- [x] List (#1390)
- [x] Get (#1391)
- [ ] Update
- [ ] Delete

[API Reference](https://developer.openstack.org/api-ref/network/v2/?expanded=#networking-agents)

### PR DESCRIPTION
**Pull request by kayrus**
Resolves #1389

https://github.com/openstack/neutron/blob/0cf41c75d50aa71d13f7391bbb91834d96d2068e/neutron/db/models/agent.py#L57-L58

https://github.com/openstack/neutron/blob/0cf41c75d50aa71d13f7391bbb91834d96d2068e/neutron/extensions/dhcpagentscheduler.py#L42-L63

### CODE REVIEW

**Comment by kayrus** [comment]
@jtopjian ready for review

File: openstack/networking/v2/extensions/agents/requests.go [hunk 1]
```diff
@@ -66,10 +66,86 @@ func Get(c *gophercloud.ServiceClient, id string) (r GetResult) {
...
 func ListDHCPNetworks(c *gophercloud.ServiceClient, id string) (r ListDHCPNetworksResult) {
 	resp, err := c.Get(listDHCPNetworksURL(c, id), &r.Body, nil)
 	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
 	return
 }
+
+// ScheduleDHCPNetworkOptsBuilder allows extensions to add additional parameters
+// to the ScheduleDHCPNetwork request.
+type ScheduleDHCPNetworkOptsBuilder interface {
+	ToScheduleDHCPNetworkMap() (map[string]interface{}, error)
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/agents/requests.go:118]: ToAgentScheduleDHCPNetworkMap

Comment (kayrus) [reply openstack/networking/v2/extensions/agents/requests.go]: fixed

**Comment by kayrus** [comment]
@jtopjian renamed the func as suggested

**Pull request merged**